docs/the-new-architecture/pillars-codegen.md

```diff
@@ -3,5 +3,227 @@ id: pillars-codegen
+import Tabs from '@theme/Tabs'; import TabItem from '@theme/TabItem'; import constants from '@site/core/TabsConstants';
+
+The **CodeGen** is not a proper pillar, but it is a tool we developed to avoid writing of a lot of boilerplate code. Using the **CodeGen** is not mandatory: all the code that is generated by it can also be written manually. However, it generates several classes that could save a lot of time.
+
+The **CodeGen** is invoked automatically by React Native every time we build an iOS or an Android app. There are cases when we would like to generate the code manually to know a type or of a file name that we need: this is a common scenario when developing **TurboModules** and **Fabric Components**, for example.
+
+This guide teaches how to configure the **CodeGen**, how to invoke it manually for each platform, and it describes the generated code.
+
+# Prerequisites
+
+We always need a React Native app to generate the code properly, even if we are invoking the **CodeGen** manually.
+
+The **CodeGen** process is tightly coupled with the build of the app and we are going to use some scripts that are located in the `react_native` Node module. Therefore, make sure to create a sample app using the command:
+
+```sh
+npx react-native init SampleApp --version 0.70.0
+```
+
+:::note
+This guide assumes that we are using a version of React Native that is 0.70.0 or greater.
+Previous versions of React Native uses a **CodeGen** that requires a slightly different setup.
+:::
+
+Then, we need to add the module that requires the **CodeGen** as an npm dependency of the app. This can be done with the command:
+
+```sh
+yarn add <path/to/your/TurboModule_or_FabricComponent>
+```
+
+See how to create a [TurboModule](./pillars-turbomodule) or a [Fabric Component](./pillar-fabric-components) to get more information on how to configure them.
+
+The rest of this guide assumes that you have a `TurboModule` and/or a `Fabric Component` properly set up.
+
+# iOS
+
+## Running the CodeGen
+
+The **CodeGen** for iOS relies on some Node scripts that are invoked during the build process. The scripts are located in the `MyApp/node_modules/react_native/scripts/` folder.
+
+The script to run is the `generate-artifacts.js` script. This searches among all the dependencies of the app, looking for JS files which respects some specific conventions (look at [TurboModules](./pillars-turbomodule) and [Fabric Components](./pillar-fabric-components) sections for details) and generates the required code.
+
+To invoke the script we can run this command from the root folder of our app:
+
+```sh
+node node_modules/react_native/scripts/generate-artifacts.js \
+    --path MyApp/ \
+    --outputPath <an/output/path> \
+```
+
+Given that the app has `TurboModules` and/or `Fabric Components` configured as a dependency, the **CodeGen** will look for all of them and it will generate the code in the `an/outputh/path` path.
```

Comment:
```suggestion
Given that the app has `TurboModules` and/or `Fabric Components` configured as a dependency, the **CodeGen** will look for all of them and it will generate the code in the path you provided.
```